src/network/peers.rs

```diff
@@ -1043,8 +1056,14 @@ impl<N: Network, E: Environment> Peer<N, E> {
                                     peer.node_type = node_type;
                                     // Update the status of the peer.
                                     peer.status.update(status);
-                                    // Update the block header of the peer.
-                                    peer.block_header = block_header;
+
```

Comment:
it can be noted here that this deserialization is deferred and non-blocking, so it's clear why it happens here